### PR TITLE
Project controller test part 4

### PR DESCRIPTION
**Pull request by mdeniz**
Some cleanup and migration of test to Rspec

### CODE REVIEW

File: src/api/spec/controllers/webui/project_controller_spec.rb [hunk 1]
```diff
@@ -272,13 +272,12 @@
   end
 
   describe 'GET #new_package_branch' do
-    before do
+    it do
```
Comment (bgeuken) [inline src/api/spec/controllers/webui/project_controller_spec.rb]: Please add a string that describes the expectation of the test

File: src/api/app/controllers/webui/project_controller.rb [hunk 2]
```diff
@@ -239,8 +239,39 @@ def add_group
```
Comment (bgeuken) [inline src/api/app/controllers/webui/project_controller.rb]: There is a typo in the commit message.

File: src/api/app/controllers/webui/project_controller.rb [hunk 3]
```diff
@@ -974,39 +1000,6 @@ def find_maintenance_infos
...
-        counts = Array.new
-      end
-      oarch = arch
-      counts << [Buildresult.index2code(code), count]
-    end
-    archs << [oarch, counts] if oarch
-    repos << [orepo, archs] if orepo
-    @buildresult = repos || Array.new
-  end
-
```
Comment (bgeuken) [inline src/api/app/controllers/webui/project_controller.rb]: Did you change the code at all or just moved it over?

File: src/api/app/controllers/webui/user_controller.rb [hunk 4]
```diff
@@ -188,8 +188,8 @@ def register
     else
       session[:login] = opts[:login]
       User.current = User.find_by_login(session[:login])
-      if Project.where(name: User.current.home_project_name).exists?
-        redirect_to project_show_path(User.current.home_project_name)
+      if User.current.home_project
```
Comment (bgeuken) [inline src/api/app/controllers/webui/user_controller.rb]: Great idea:+1:

File: src/api/app/controllers/webui/project_controller.rb (hunk 3 above)
Comment (mdeniz) [inline src/api/app/controllers/webui/project_controller.rb]: Moved it. It was "splited" in 3 methods, 2 of them only used in the real action

File: src/api/app/controllers/webui/user_controller.rb (hunk 4 above)
Comment (mdeniz) [inline src/api/app/controllers/webui/user_controller.rb]: :smile:

File: src/api/app/controllers/webui/project_controller.rb (hunk 2 above)
Comment (mdeniz) [inline src/api/app/controllers/webui/project_controller.rb]: What typo? In what commit? You mean in this one? "[webui] Refactor ProjecController#buildresult"

Comment (bgeuken) [inline src/api/app/controllers/webui/project_controller.rb]: Yep, that one. Couldn't comment on the whole commit. That's why I did it here.

**Pull request merged**